_posts/2016-02-03-variance-and-functors.md

```diff
@@ -0,0 +1,475 @@
+---
+layout: post
+title: Of variance and functors
+
+meta:
+  nav: blog
+  author: adelbertc
+  pygments: true
+---
+Scala's type system allows us to annotate type parameters with their variance: covariant, contravariant, invariant.
+Variance allows us to define the subtyping relationships between type constructors - that is, under which
+conditions `F[A]` is a subtype of `F[B]`.
+
+Similarly in functional programming, there are covariant functors, contravariant functors, and invariant functors. The
+similarity in names is not coincidental.
+
+# Covariance
+The common example is `List[+A]` which is covariant in its type parameter, denoted by the `+` next to the `A`.
+A type constructor with a covariant type parameter means that if there is a subtyping relationship between the
+type parameter, there is a subtyping relationship between the two instances of the type constructor. This means that
```

Comment:
s/This means that/For example/

("means that" already appears in the previous sentence)